lib/instrumentation/span.js

```diff
@@ -125,6 +124,18 @@ Span.prototype.setMessageContext = function (context) {
   this._message = Object.assign(this._message || {}, context)
 }
 
+Span.prototype.shouldSetSpanStackTrace = function () {
+  if (this._agent._conf.spanFramesMinDuration === -1) {
```

Comment:
The [Java agent docs](https://www.elastic.co/guide/en/apm/agent/java/current/config-stacktrace.html#config-span-frames-min-duration) (and its [impl](https://github.com/elastic/apm-agent-java/blob/a837d43aa3fa189c19ae5dadaa4e0e28ae52d97a/apm-agent-core/src/main/java/co/elastic/apm/agent/impl/ElasticApmTracer.java#L397-L399)) say:

> When setting this option to a negative value, like -1ms, stack traces will be collected for all spans. 

In other words, it isn't `-1` specifically that is a special case; it is *any* negative value.  And actually it doesn't need to be special-cased because any `duration` will be greater than a negative value.  So the implementation above could be simplified to:

```
if (this._capturedStackTrace !== null && this._agent._conf.spanFramesMinDuration !== 0 && this.duration() / 1000 > this._agent._conf.spanFramesMinDuration) { ... }
```

and potentially not have the separate `.shouldSetSpanStackTrace()` method -- though that is fine too.

Reply:
fair -- I might argue the separate method makes the intent of the long conditional clearer, but we don't have any linting rules against those long conditionals to I'll let that lie ;)   Dropped the above code in place and tests (plus a new one with a spanFramesMinDuration: '-2' are passing.